python/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+if (PYTHON_EXECUTABLE)
+    set(Python3_EXECUTABLE ${PYTHON_EXECUTABLE})
```

Comment:
Usually we should just be able to bank on the `Python3_EXECUTABLE` variable but `python -m build` provides a separate venv python executable for just the building/packaging process. If it's set, we'll use it, otherwise the `Python3_EXECUTABLE` should be used.